File: base/core/common/db/orm/menu_item.rb
require 'securerandom'
class MenuItem < Sequel::Model
  many_to_one :parent, class: self
  one_to_many :children, key: :parent_id, class: self
  def before_create
    self.uuid = SecureRandom.uuid
  end
  def before_destroy
    self.children.each do |c|
      c.destroy
    end
  end
end

File: base/quasar/websocket/src/stores/orm/MenuItem.js
import { Model } from 'pinia-orm'

export default class MenuItem extends Model {

  static entity = 'menu_items'
  static primaryKey = 'uuid'

  static fields () {
    return {
      uuid: this.string(null),
      parent_id: this.string(null),
      uri: this.string(null),
      caption: this.string(null),
      subtitle: this.string(null),
      icon: this.string(null),
      sorted_order: this.number(null),
      public: this.boolean(null),
      children: this.hasMany(MenuItem, 'parent_id'),
      parent: this.belongsTo(MenuItem, 'parent_id')
    }
  }
}
